### PR TITLE
🤘 emoji list

### PR DESCRIPTION
**Pull request by lisatsewang**
Screen Shot 2019-08-15 at 2 23 13 PM

### CODE REVIEW

File: emoji-ui/src/components/sourcelist/sourcelist.js [hunk 1]
```diff
@@ -27,8 +28,8 @@ class SourceListBase extends Component {
 
   render() {
     const { sourceEmoji } = this.props;
-    const firstTwenty = sourceEmoji.slice(0, 20);
-    return <h1>{JSON.stringify(firstTwenty)}</h1>;
+    const firstBatch = sourceEmoji.slice(0, 200);
```
Comment (gmkohler) [inline emoji-ui/src/components/sourcelist/sourcelist.js]: Are we planning on adding pagination on the front or backend?

Comment (lisatsewang) [reply emoji-ui/src/components/sourcelist/sourcelist.js]: yea, i am thinking about working on pagination next if no one is going to work on it

File: emoji-ui/src/components/emojiItem.js [hunk 2]
```diff
@@ -4,14 +4,17 @@ import './emojiItem.css';
 
 class EmojiItem extends Component {
   static propTypes = {
-    isSelected: PropTypes.bool.isRequired,
-    url: PropTypes.string.isRequired,
```
Comment (clairewild) [inline emoji-ui/src/components/emojiItem.js]: Why'd you make the props no longer required?

Comment (lisatsewang) [reply emoji-ui/src/components/emojiItem.js]: i am thinking if an emoji is already imported, maybe they don't need a click handler.

Comment (clanam) [reply emoji-ui/src/components/emojiItem.js]: in general default always to using isRequired unless you already know it won’t always be sent
you’ll get much better feedback in the console if there are bugs passing the wrong things
and it’s easier to relax a requirement than to remember to enforce an optional one

File: emoji-ui/src/components/emojiList.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        name: PropTypes.string,
+        url: PropTypes.string,
+      })
+    ),
+    clickHandler: PropTypes.func.isRequired,
+  };
+
+  render() {
+    const { sourceEmoji } = this.props;
+    const emojiListItem = sourceEmoji.map((emoji, idx) => {
```
Comment (clanam) [inline emoji-ui/src/components/emojiList.js]: `key={idx} emoji={emoji}`  — much better is `key={emoji.name} emoji={emoji} `
the key is how react figures out which dom objects are identical for the purpose of minimizing dom-changes in re-rendering
so if you have a key that id’s an object you want to use that always
in this case, name identifies an emoji far more than its position in the list (esp. if we add search/filter)
so passing name as the key gives optimal diffing

File: emoji-ui/src/components/emojiItem.js (hunk 2 above)
Comment (lisatsewang) [reply emoji-ui/src/components/emojiItem.js]: oh @clairewild i missunderstood your comment. yes, the emoji needs to be required. it was a mistake. i will update that.

the clickHandler will remain optional.

File: emoji-ui/src/components/emojiList.js (hunk 3 above)
Comment (lisatsewang) [reply emoji-ui/src/components/emojiList.js]: sounds good, will update!

**Pull request merged**